Rework persistent offloading based on upstream

crawfxrd opened the pull request:
fd.o has accepted the name `PrefersNonDefaultGPU` [\[1\]] and Bastien is working on integrating it into gnome-shell [\[2\]].

- Use `PrefersNonDefaultGPU` instead of `X-KDE-RunOnDiscreteGpu`
- Preserve `shell_app_launch()` signature for extensions (Resolves: #26)

[\[1\]]: https://gitlab.freedesktop.org/xdg/xdg-specs/-/merge_requests/13
[\[2\]]: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1226

jackpot51 commented:
Don't want to make this too complicated, but is there a reason not to check for both? Is X-KDE-RunOnDiscreteGpu deprecated?

crawfxrd commented:
`PrefersNonDefaultGPU` is the standardized name for `X-KDE-RunOnDiscreteGpu`. GNOME is implementing the feature only after having the entry added to the XDG spec, so they have no reason to check for the KDE-specific name.

Applications used across multiple desktop environments should probably check for both. Such as what Bastien did in [xdg/desktop-file-utils!7](https://gitlab.freedesktop.org/xdg/desktop-file-utils/-/merge_requests/7).

crawfxrd commented:
They've decided to use "Integrated" for the user visible text ("Default" too similar to "Dedicated"), so I've changed it back to that.

crawfxrd committed:
Rework persistent offloading based on upstream

fd.o has accepted the name PrefersNonDefaultGPU [1] and Bastien is
working on integrating it into gnome-shell [2].

[1]: https://gitlab.freedesktop.org/xdg/xdg-specs/-/merge_requests/13
[2]: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1226

crawfxrd commented:
Updated to preserve the signature of `shell_app_launch()` from GNOME 3.36 so extensions can use it without breaking.
